### PR TITLE
Task/oracle test

### PR DESCRIPTION
**Pull request by antoncoding**
## Contract Changes
* Separate `CompoundOracleInterface` and `OracleInterface`: `CompoundOracleInterface` is the one `Oracle.sol` use to interact with compound v1 oracle, and `OracleInterface` is just the interface for `Oracle.sol`
* Add `MockCompoundOracle` and `MockOracle` accordingly, use `MockCompoundOracle` in `Oracle`'s unit test.
* Add setter function for priceOracle. Owner can update the address of compoundOracle. 
* Change variable name in `OptionsContract.sol` from `compoundOracle` to `oracle`
* Remove` OptionExchange` in `OptionsContract`, declare that variable in `Otoken`, which is the only one using it.

### CODE REVIEW

File: contracts/oToken.sol [hunk 1]
```diff
@@ -9,6 +10,8 @@ import "./OptionsContract.sol";
  */
 
 contract oToken is OptionsContract {
+    OptionsExchange public optionsExchange;
+
```
Comment (aparnakr) [inline contracts/oToken.sol:14]: hmm why is this here as opposed to the previous verison?

Comment (antoncoding) [reply contracts/oToken.sol:14]: Because the variable is only used in `oToken.sol`, so I moved it here and remove from the constructor of `OptionsContract.sol` so silence some warning from Crytic

File: test/oracle.test.ts [hunk 2]
```diff
@@ -1,120 +1,219 @@
...
+      bat = await MockERC20.new('BAT', 'BAT', 18);
+      await oracle.setBat(bat.address, {from: owner});
+      const cBatToBatExchangeRate = '203779026431652476585639266'; // 0.023779 * 1e28
+      cBat = await MockCToken.new(bat.address, cBatToBatExchangeRate);
+      await oracle.setCbat(cBat.address, {from: owner});
+      await oracle.setIsCtoken(cBat.address, true);
+      await oracle.setAssetToCtoken(bat.address, cBat.address);
+
+      // update compound oracle price for BAT
+      const batPrice = '777857500000000';
```
Comment (aparnakr) [inline test/oracle.test.ts:155]: Is it possible to use ganache fork for this and also test with mainnet data? (Also these tests will probs change since we need to use a different oracle now right?)

Comment (antoncoding) [reply test/oracle.test.ts:155]: This is the data I copied from mainnet, for unit tests I think it's okay that we just hardcode the value, because we don't have existing cicd workflow to run ganache fork now.
